kernel/kernel-keymanager-service/src/main/java/io/mosip/kernel/keymanagerservice/helper/SessionKeyDecrytorHelper.java

```diff
@@ -117,7 +117,8 @@ public SymmetricKeyResponseDto decryptSessionKey(SymmetricKeyRequestDto symmetri
 				
 		byte[] encryptedData = CryptoUtil.decodeURLSafeBase64(symmetricKeyRequestDto.getEncryptedSymmetricKey());
 
-		if (noThumbprint) {
+		if (noThumbprint || (encryptedData.length != (CryptomanagerConstant.ENCRYPTED_SESSION_KEY_LENGTH 
```

Comment:
For decryption why do we need noThumbprint property ? Hope this change will fix above question 3 ?

Reply:
yes, after this fix, it will not have any affect now on any module.